### ISSUE TITLE
False `Too many arguments to def` errors with `:macroexpand` hooks

### ISSUE THREAD

**Issue by camsaul** [clj-kondo/clj-kondo#2026]
**version**

```
clj-kondo v2023.03.17
```

**macro usage**

I have a `:macroexpand` hook like

```clj
(defmacro defmodel
  [model _table-name]
  `(def ~model "Docstring." ~(keyword (name model))))
```

for macros like

```clj
(t1.models/defmodel MyCategoryDefModel :t1_category)
```

and this previously worked fine (I was using `v2023.01.something`), but after upgrading to `v2023.03.17` I am getting errors like 

```
toucan1/test/toucan/models_test.clj:36:1: error: Too many arguments to def
```

These seem to be false positives.

**platform**

Native version installed with Homebrew (Linux)

**repro**

You can repro this by cloning https://github.com/camsaul/toucan2, commit `5e92bc31b59a9b3ec36adae206096becce93c9ea`, and running 

```sh
clj-kondo --lint toucan1/test
```
